### ISSUE TITLE
Using assert_type on type parameter of a generic: Unknown vs. Any

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#3794]
We're trying something with generics in pandas-stubs, and have noticed a difference in behavior with `pyright` and `mypy` that we'd like to reconcile.

Consider this type stub, called `genericret.pyi`:
```python
from __future__ import annotations
from typing import Generic, TypeVar, overload

S1 = TypeVar("S1", int, str)

class Foo(Generic[S1]):
    @overload
    def __new__(cls, dtype: type[S1]) -> Foo[S1]: ...
    @overload
    def __new__(cls, dtype=...) -> Foo: ...
    def func(self) -> S1: ...
```
And this test python code `genrectst.py`:
```python
from typing_extensions import assert_type, reveal_type

from genericret import Foo
from typing import Any

f = Foo(int)
reveal_type(f)
reveal_type(f.func())

g = Foo()
reveal_type(g)
reveal_type(g.func())

assert_type(g.func(), Any)
```

In this case, if a `dtype` is specified, we can say that we know the specific type of `Foo`.  But if `dtype` is not specified, then it is unknown.  But that's OK - except we want the function `func()` to then return `Any` as a type, which is what `mypy` does. 

`mypy` output:
```text
genrectst.py:7: note: Revealed type is "genericret.Foo[builtins.int]"
genrectst.py:8: note: Revealed type is "builtins.int"
genrectst.py:11: note: Revealed type is "genericret.Foo[Any]"
genrectst.py:12: note: Revealed type is "Any"
Success: no issues found in 1 source file
```

`pyright` output:
```text
  genrectst.py:7:13 - information: Type of "f" is "Foo[int]"
  genrectst.py:8:13 - information: Type of "f.func()" is "int"
  genrectst.py:11:13 - information: Type of "g" is "Foo[Unknown]"
  genrectst.py:12:13 - information: Type of "g.func()" is "Unknown"
  genrectst.py:14:13 - error: "assert_type" mismatch: expected "Any" but received "Unknown" (reportGeneralTypeIssues)
```

So with `mypy`, we can use `assert_type()` to check the result of `func()`, but how do we do this with `pyright` ?

**Comment by erictraut** [comment]
`Unknown` is pyright's way of tracking an "implied Any", so it should evaluate to the same as `Any` in an `assert_type` statement.

I'll make that change so the assert doesn't generate an error in the above code.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.266, which I just published. It will also be included in a future release of pylance.
